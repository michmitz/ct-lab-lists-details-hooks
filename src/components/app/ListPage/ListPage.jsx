import React from 'react';
import PropTypes from 'prop-types';
import DetailPage from '../DetailPage/DetailPage';

const ListPage = ({ characters }) => {
  const characterlist = characters.map(character => (
    <a key={character.name} href={`/${character.name}`}>
           <DetailPage {...character} />
    </a>
  ));

  return (
    <ul data-testid="character-list">
      {characterlist}
    </ul>
  );
};

ListPage.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired
};

export default ListPage;
