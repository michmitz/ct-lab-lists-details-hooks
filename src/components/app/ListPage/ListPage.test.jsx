import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ListPage from './ListPage';

describe('ListPage component', () => {
  afterEach(() => cleanup());
  it('renders ListPage', () => {
    const { asFragment } = render(<ListPage
      characters={[
        {
          name: 'Michelle',
          description: 'They cut her scenes from X-Files, but she was there',
          image: 'http://placeholder.com/michelle'
        }
      ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});