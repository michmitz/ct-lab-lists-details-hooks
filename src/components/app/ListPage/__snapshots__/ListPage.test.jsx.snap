// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage component renders ListPage 1`] = `
<DocumentFragment>
  <ul
    data-testid="character-list"
  >
    <a
      href="/Michelle"
    >
      <div
        data-testid="character-details"
      >
        <h2>
          Michelle
        </h2>
        <p>
          They cut her scenes from X-Files, but she was there
        </p>
        <img
          src="http://placeholder.com/michelle"
        />
      </div>
    </a>
  </ul>
</DocumentFragment>
`;
